File: sources/Android/android-communications/library/src/main/java/com/polar/androidcommunications/api/ble/model/offlinerecording/OfflineRecordingError.kt
package com.polar.androidcommunications.api.ble.model.offlinerecording

sealed class OfflineRecordingError(detailMessage: String) : Exception(detailMessage) {
    object OfflineRecordingEmptyFile : OfflineRecordingError("")
    object OfflineRecordingNoPayloadData : OfflineRecordingError("")
    object OfflineRecordingHasWrongSignature : OfflineRecordingError("")
    object OfflineRecordingErrorSecretMissing : OfflineRecordingError("")
    object OfflineRecordingErrorNoParserForData : OfflineRecordingError("")
    class OfflineRecordingSecurityStrategyMissMatch(detailMessage: String) : OfflineRecordingError(detailMessage)
    class OfflineRecordingErrorMetaDataParseFailed(detailMessage: String) : OfflineRecordingError(detailMessage)
}

File: sources/iOS/ios-communications/Sources/iOSCommunications/ble/api/model/offlinerecording/OfflineRecordingError.swift
//  Copyright © 2022 Polar. All rights reserved.

import Foundation

/// Offline recording errors
public enum OfflineRecordingError: Error {
    case emptyFile
    case offlineRecordingErrorMetaDataParseFailed(description: String = "")
    case offlineRecordingErrorSecretMissing
    case offlineRecordingSecurityStrategyMissMatch(description: String = "")
    case offlineRecordingHasWrongSignature
    case offlineRecordingNoPayloadData
    case offlineRecordingErrorNoParserForData
}
